messaging/src/test/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessorModuleTest.java

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.pooled;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import org.axonframework.common.transaction.NoOpTransactionManager;
+import org.axonframework.configuration.AxonConfiguration;
+import org.axonframework.configuration.Configuration;
+import org.axonframework.configuration.MessagingConfigurer;
+import org.axonframework.eventhandling.*;
+import org.axonframework.eventhandling.configuration.EventHandlingComponentsConfigurer;
+import org.axonframework.eventhandling.configuration.EventProcessorModule;
+import org.axonframework.eventhandling.tokenstore.TokenStore;
+import org.axonframework.eventhandling.tokenstore.inmemory.InMemoryTokenStore;
+import org.axonframework.eventstreaming.StreamableEventSource;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.SimpleUnitOfWorkFactory;
+import org.axonframework.messaging.unitofwork.TransactionalUnitOfWorkFactory;
+import org.axonframework.messaging.unitofwork.UnitOfWorkFactory;
+import org.axonframework.utils.AsyncInMemoryStreamableEventSource;
+import org.junit.jupiter.api.*;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiFunction;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+/**
+ * Test class validating the {@link PooledStreamingEventProcessorModule} functionality and its integration with the {@link MessagingConfigurer}.
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+class PooledStreamingEventProcessorModuleTest {
+
+    @Nested
+    class ConstructionTest {
+
+        @Test
+        void shouldCreateModuleWithProcessorName() {
+            // given
+            String processorName = "test-processor";
+
+            // when
+            PooledStreamingEventProcessorModule module = EventProcessorModule
+                    .pooledStreaming(processorName)
+                    .eventHandlingComponents((cfg, components) -> components.single(new SimpleEventHandlingComponent()))
```

Comment:
I think to support annotated/declarative components there will be some changes, like below - but in the following PRs. 


```suggestion
                    .eventHandlingComponents((cfg, components) -> components
                           .annotated(new SimpleEventHandlingComponent()
                           .declarative(...)
                     )
```